### PR TITLE
Quickstarts Q3 2023 update

### PR DESCRIPTION
**Pull request by john-rock**
## What are you changing in this pull request and why?

[Asana task](https://app.asana.com/0/1200099998847559/1205319420559417/f)

This PR introduces enhancements to the Quickstart layout, providing users with a more powerful and intuitive experience. Here's what's new:

1. **Filter by Topic** 🧩
   - Users can now filter quickstarts by topics of interest.
   - Introduces the `tags` markdown property on quickstarts as an array where users can specify the desired topics for filtering.

2. **Filter by Level** 📊
   - Quickstarts can be filtered based on their complexity level.
   - The `level` property, a string, allows users to set and filter by the desired level.

3. **Search Bar** 🔍
   - A convenient search bar has been added to enable users to quickly find specific quickstarts.

Additionally, added the `recently_updated` property, a boolean which when set to true will display an "updated" label on quickstarts that have been recently modified.
[
Quickstart notion doc](https://www.notion.so/dbtlabs/Quickstart-Guides-73c57db4b37547aaab4b7d9437e19eb1) has been updated as well outlining the features above.

## Preview Link
https://docs-getdbt-com-git-quickstarts-q32023-update-dbt-labs.vercel.app/guides

## Loom Walkthrough
https://www.loom.com/share/bce7a1b7ec944e109c0f251bd8566dc3

## Next Steps
This PR adds sample data to each quickstart `tags`, `level`, and `recently_updated` properties in order to test the functionality. A seperate PR will need to be opened applying the correct information on each quickstart.

Any work involved with the [migration](https://www.notion.so/dbtlabs/Quickstarts-Q3-updates-SOW-b36903d2fffc4320b27a594017010214?pvs=4#b6e72839f3364923b1da7b4b61fb3090) part of this task can begin to be worked on.

## Must be completed before merging
- [x] `/quickstarts` directory changed to `/guides`
- [x] Quickstarts updated with correct tagging
- [x] Migration work completed

### CODE REVIEW

File: website/src/components/quickstartGuideList/index.js [hunk 1]
```diff
@@ -1,19 +1,68 @@
...
+  // Get tag options
+  // Populated from the tags frontmatter array
+  const tagOptions = useMemo(() => {
+    const tags = new Set();
+    quickstartData.forEach(guide =>
+      guide?.data?.tags?.forEach(tag => tags.add(tag))
+    );
+    // Sort alphabetically
+    return Array.from(tags).sort((a, b) => a.toLowerCase().localeCompare(b.toLowerCase())).map(tag => ({ value: tag, label: tag }));
+  }, [quickstartData]);
```
Comment (JKarlavige) [inline website/src/components/quickstartGuideList/index.js:36]: Would be interested in diving into this more as i'm not too familiar with `useMemo`!

File: website/src/components/quickstartTOC/styles.module.css [hunk 2]
```diff
@@ -88,10 +90,20 @@ html[data-theme="dark"] .tocList .active span {
...
+
 .buttonContainer .nextButton {
     margin-left: auto;
 }
 
+.buttonContainer .nextButton i {
+    font-size: .8rem;
+    margin-left: .4rem;
+}
+
```
Comment (JKarlavige) [inline website/src/components/quickstartTOC/styles.module.css:110]: When on step 1 of a quickstart, the `Next` button at the bottom of the page has green text with a green button when in dark mode. Once you move to step 2 and on, the button has a dark bg and is readable.
image

File: website/src/components/selectDropdown/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+      name="multi-select-dropdown"
+      options={options}
+      className={styles.selectContainer}
+      classNamePrefix={styles.select}
+      value={value}
+      onChange={onChange}
+      placeholder={placeHolder}
+      aria-label="Multi Select Dropdown"
+      unstyled
+    />
```
Comment (JKarlavige) [inline website/src/components/selectDropdown/index.js:18]: This component looks and works great!

**Review by JKarlavige — CHANGES_REQUESTED** [review]
Dang this came out awesome, great work! All filtering/searching looks to function correctly. Just have one small requested change for the `Next` button on the individual quickstart pages.

**Comment by john-rock** [comment]
Recent commits include updating `/quickstarts` to `/guides` per @runleonarun request in the [task](https://app.asana.com/0/0/1205319420559417/1205817625817418/f).

File: website/docs/quickstarts/bigquery-qs.md
Comment (JKarlavige) [inline website/docs/quickstarts/bigquery-qs.md:1]: Is this bigquery file within the quickstarts safe to be deleted?

File: website/docusaurus.config.js [hunk 4]
```diff
@@ -134,8 +134,8 @@ var siteSettings = {
               to: '/guides/best-practices',
             },
             {
-              label: "Quickstarts",
-              to: "/quickstarts",
+              label: "Guides",
+              to: "/guides",
```
Comment (JKarlavige) [inline website/docusaurus.config.js:138]: Right above this nav item there is another `Guides` nav item which links to `/guides/best-practices`.  Thinking we'll have to rename this existing nav item.
image

**Review by JKarlavige — CHANGES_REQUESTED** [review]
Just have one requested change in the nav, as there are now two `Guides` nav items. Clicked through all guides and looks to be functioning correctly!

File: website/docusaurus.config.js (hunk 4 above)
Comment (john-rock) [reply website/docusaurus.config.js:138]: Ope

Comment (john-rock) [reply website/docusaurus.config.js:138]: @runleonarun do you have a preference on what we want to rename/replace this with?

Comment (runleonarun) [reply website/docusaurus.config.js:138]: @john-rock That should be "Best practices" for now and everything in that best-practices folder should live in website/docs/best-practices/

Comment (john-rock) [reply website/docusaurus.config.js:138]: Got it thanks @runleonarun! That's updated now.

**Review by JKarlavige — APPROVED** [review]
lgtm! :shipit:

**Comment by runleonarun** [comment]
@JKarlavige  and @john-rock I'm going to go ahead and try to fix the merge conflict. 🤞🏻

**Comment by john-rock** [comment]
Awesome thanks @runleonarun! With this approved you should be good to move forward with the migration process.

**Comment by runleonarun** [comment]
Note that the content work is being done [over in this PR](https://github.com/dbt-labs/docs.getdbt.com/pull/4361/files)

**Comment by runleonarun** [comment]
@john-rock I'm curious about the front matter `platform: 'dbt-core'` It seems redundant to the `tags: ['migration', 'dbt Core','dbt Cloud']`. Or is there a way to sort by platform? Thank you! Sorry if I missed this.

**Comment by john-rock** [comment]
> @john-rock I'm curious about the front matter `platform: 'dbt-core'` It seems redundant to the `tags: ['migration', 'dbt Core','dbt Cloud']`. Or is there a way to sort by platform? Thank you! Sorry if I missed this.

Hey @runleonarun great call out!

The `platform` front matter was setup during V1 of this project in anticipation for V2. We went with a slightly different pattern for V2 and now that we are using the tags front matter with `core` and `cloud` values the `platform` is indeed redundant. 

Feel free to drop the `platform` front matter from these!

**Comment by nghi-ly** [comment]
helping leona usher this to production while she's OOO. [slack convo](https://dbt-labs.slack.com/archives/C02NCQ9483C/p1699582238121899?thread_ts=1699567425.027689&cid=C02NCQ9483C)

**Pull request merged**